File: gemini_tests/java/PathTraversal.java

import java.io.File;
import javax.servlet.http.HttpServletRequest;

public class PathTraversal {
    public void vulnerable(HttpServletRequest request) {
        String filename = request.getParameter("filename");
        // Vulnerable to Path Traversal
        File file = new File("/var/www/images/" + filename);
    }
}

File: gemini_tests/go/path_traversal.go

package main

import (
	"io/ioutil"
	"net/http"
	"path/filepath"
)

func vulnerable(w http.ResponseWriter, r *http.Request) {
	filename := r.URL.Query().Get("file")
	// Vulnerable to Path Traversal
	data, _ := ioutil.ReadFile(filepath.Join("/var/www/", filename))
	w.Write(data)
}
